Client/Ecosia/UI/Onboarding/Welcome.swift

```diff
@@ -46,10 +47,8 @@ final class Welcome: UIViewController {
         addBackground()
         addStack()
 
-        if LegacyThemeManager.instance.systemThemeIsOn {
-            let userInterfaceStyle = traitCollection.userInterfaceStyle
-            LegacyThemeManager.instance.current = userInterfaceStyle == .dark ? LegacyDarkTheme() : LegacyNormalTheme()
-        }
+        let themeManager: ThemeManager = AppContainer.shared.resolve()
+        (themeManager as? EcosiaThemeManager)?.updateLegacyThemeIfNeeded()
```

Comment:
In the end I believe `EcosiaThemeManager.updateLegacyThemeIfNeeded` and `LegacyThemeManager. updateBasedOnCurrentSystemThemeType` are very similar with the only difference being the first one check is the system theme is one.

Could we keep only one of them to avoid duplicated logic? I guess keeping the `LegacyThemeManager` one makes sense since that's the updated class but adding the system check?

Reply:
Meant to keep the system check separated as it's implicit and the function will be only responsible to update the theme, without checking the system theme within.